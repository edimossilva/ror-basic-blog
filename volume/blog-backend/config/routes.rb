Rails.application.routes.draw do
  post '/auth/login', to: 'authentication#login'
  
  resources :blogs, :only => [:create, :destroy, :show] do
    resources :posts, :only => [:create, :destroy]
  end
end
