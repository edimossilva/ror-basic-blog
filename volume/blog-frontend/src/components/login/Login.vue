<template lang="">
  <div>
    <p>Login</p>

    <div>
      <label> Username: </label>
      <input type="text" v-model="username" class="Login__username-input-js"> 
    </div>
    
    <div>
      <label> Password: </label>
      <input type="password" v-model="password" class="Login__password-input-js"> 
    </div>

    <button @click="doLogin" class="Login__login-button-js">Login</button>
    <div>
      username: 'registered_user1', password: '111' <br>
      username: 'registered_user2', password: '222' <br>
      username: 'admin_user1',      password: '111' <br>
      username: 'admin_user2',      password: '222' <br>
    </div>
  </div>
</template>

<script>
import authService from "../../services/auth_service";

export default {
  data() {
    return {
      username: "registered_user1",
      password: "111"
    };
  },
  methods: {
    doLogin() {
      const { username, password } = this;
      const { onLoginSuccess, showErrorMessage } = this;

      authService
        .doLogin(username, password)
        .then(onLoginSuccess)
        .catch(showErrorMessage);
    },

    onLoginSuccess({ data: { token, username } }) {
      this.saveToken(token);
      this.saveUsername(username);
      this.redirectToHomeIfLogged();
      console.log("redirecthome");
    },

    saveToken(token) {
      this.$store.commit("saveAuthToken", token);
    },

    saveUsername(username) {
      this.$store.commit("saveUsername", username);
    },

    showErrorMessage(error) {
      // debugger; // eslint-disable-line
      console.log(`error ${error}`);
    },
    redirectToHomeIfLogged() {
      if (this.$store.getters.isLoged) {
        return this.$router.push({ name: "blogs" });
      }
    }
  },
  mounted: function() {
    this.redirectToHomeIfLogged();
  }
};
</script>

<style scoped></style>