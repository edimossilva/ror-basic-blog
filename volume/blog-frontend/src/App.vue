<template>
  <div>
    <user-header></user-header>
    <routes-menu :routes="routes"></routes-menu>
    <router-view></router-view>
    <alert-modal></alert-modal>
  </div>
</template>

<script>
import { routes } from "./routes";
import RoutesMenu from "./components/routes-menu/RoutesMenu";
import UserHeader from "./components/user-header/UserHeader";
import AlertModal from "./components/alert-modal/AlertModal";
import notificationService from "./services/notification_service";

export default {
  name: "App",

  components: {
    AlertModal,
    RoutesMenu,
    UserHeader
  },

  data() {
    return {
      routes: routes
    };
  },

  mounted: function() {
    notificationService.createCable(this);
  }
};
</script>

<style>
</style>
