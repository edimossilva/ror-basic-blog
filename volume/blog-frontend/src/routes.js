import Login from './components/login/Login.vue'

export const routes = [
  { name: 'login', path: '/login', component: Login, title: 'Login' },
]