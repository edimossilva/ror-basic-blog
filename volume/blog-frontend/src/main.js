import Vue from 'vue'
import App from './App.vue'
import VueRouter from 'vue-router';

import { routes } from './routes';
import { store } from './store/store';
import VModal from 'vue-js-modal'

Vue.use(VModal)
Vue.use(VueRouter)

Vue.config.productionTip = false

const router = new VueRouter({ routes })

new Vue({
  render: h => h(App),
  router,
  store
}).$mount('#app')
