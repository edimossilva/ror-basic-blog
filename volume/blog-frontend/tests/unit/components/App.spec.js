import { mount, createLocalVue } from "@vue/test-utils"
import App from "@/App.vue"
import VueRouter from "vue-router"
import Login from '@/components/login/Login'
import { routes } from "../../../src/routes"

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter({ routes })

describe("App", () => {
  it('is a valid component', () => {
    const wrapper = mount(App, {
      localVue,
      router
    })

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  describe('router', () => {
    const wrapper = mount(App, {
      localVue,
      router
    })

    it("visits login", async () => {
      router.push("/login")
      await wrapper.vm.$nextTick()

      const loginComponent = wrapper.find(Login)
      expect(wrapper.find(Login).exists()).toBe(true)
    })
  });
})