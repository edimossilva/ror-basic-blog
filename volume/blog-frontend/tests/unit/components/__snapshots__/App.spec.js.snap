// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is a valid component 1`] = `
<div>
  <nav>
    <ul>
      <li><a href="#/login" class="router-link">Login</a></li>
      <li><a href="#/blogs" class="router-link">List Blogs</a></li>
      <li><a href="#/blogs/new" class="router-link">Create Blog</a></li>
    </ul>
  </nav>
  <!---->
</div>
`;
